C23_WMDE_Mobile_EN_02

moiikana opened the pull request:
- based on the variant banner of mobile-en-01
- var redirects to payment providers
TODO:
- [x] add tests for direct redirect

https://phabricator.wikimedia.org/T352638

gbirke commented on mobile_english_preact/components/BannerText.jsx:
```diff
@@ -0,0 +1,28 @@
+		<p>
+			It's a little awkward, so we'll get straight to the point: { weekdayPrepPhrase } { currentDayName } we humbly
+			ask you to protect Wikipedia's independence.
+			{ campaignDaySentence.getSentence() } We depend on donations averaging about € 23.83, but 99% of our readers don't give.
```
See #310 for the changed `getSentence` code
```suggestion
			{ campaignDaySentence } We depend on donations averaging about € 23.83, but 99% of our readers don't give.
```

gbirke commented on mobile_english_preact/components/FollowupTransition.jsx:
```diff
@@ -0,0 +1,158 @@
+ * by the vanished mini banner, then it must look like it's "pushing down"
+ * the rest of the page, while maintaining an overall smooth animation style.
+ */
+export default class BannerTransition extends Component {
```
Is this transition different than the one in Mobile? If not, please move it to shared instead.

gbirke commented on mobile_english_preact/styles/SelectGroup.pcss:
```diff
@@ -0,0 +1,177 @@
+
+	/* Individual button sizes for each section */
+
+	.select-group-container--interval {
```
Please add the `select-` prefix from #310 to the 3 container selectors

gbirke commented on pad_english_preact/components/BannerText.jsx:
```diff
@@ -11,12 +11,13 @@ export default function BannerText( props ) {
-			<span> { weekdayPrepPhrase } { currentDayName } we humbly ask you to protect Wikipedia's independence.
-				{ campaignDaySentence.getSentence() } We depend on donations averaging about €&nbsp;23.83, but 99% of our readers
+			<span> { weekdayPrepPhrase } { currentDayName } we humbly ask you to protect Wikipedia's independence.</span>
+			<span> { campaignDaySentence.getSentence() } We depend on donations averaging about €&nbsp;23.83, but 99% of our readers
```
```suggestion
			<span> { campaignDaySentence } We depend on donations averaging about €&nbsp;23.83, but 99% of our readers
```

moiikana replied:
yes to the function call removal, 
but that "<span> " needs to stay there for having a space before the campaigndaysentence (because preact autoremoves them).

gbirke commented on pad_english_preact/components/BannerText_var.jsx:
```diff
@@ -11,12 +11,13 @@ export default function BannerText( props ) {
-			<span> { weekdayPrepPhrase } { currentDayName } we humbly ask you to protect Wikipedia's independence.
-				{ campaignDaySentence.getSentence() } We depend on donations averaging about €&nbsp;23.83, but 99% of our readers
+			<span> { weekdayPrepPhrase } { currentDayName } we humbly ask you to protect Wikipedia's independence.</span>
+			<span> { campaignDaySentence.getSentence() } We depend on donations averaging about €&nbsp;23.83, but 99% of our readers
```
```suggestion
			<span> { campaignDaySentence } We depend on donations averaging about €&nbsp;23.83, but 99% of our readers
```

gbirke commented on mobile_english_preact/components/MiniBanner.jsx:
```diff
@@ -0,0 +1,56 @@
+					formatters={props.formatters}
+					bannerText={props.slides}
+					propsForText={{
+						formattedGoalDonationSumNumeric: props.formatters.millionFormatterNumericOnly( campaignProjection.goalDonationSum / 1000000 ),
```
IMHO the JSX fieles should contain as few logic as possible. Withe formatters (and esp. with a name like `millionFormatterNumericOnly`) it makes sense to move the division into thze formatter.

gbirke commented on shared/number_formatter/de.js:
```diff
@@ -4,5 +4,6 @@ import amountForServerFormatter from './server';
 export const integerFormatter = formatter( { round: 0, integerSeparator: '.' } );
 export const amountInputFormatter = formatter( { round: 2, suffix: ' €', decimal: ',', integerSeparator: '' } );
 export const millionFormatter = formatter( { round: 1, decimal: ',', suffix: ' Mio. €', padRight: 1 } );
+export const millionFormatterNumericOnly = formatter( { round: 1, decimal: ',', padRight: 1 } );
```
```suggestion
export const millionFormatterNumeric = formatter( { round: 1, decimal: ',', padRight: 1 } );
```

gbirke commented on shared/number_formatter/en.js:
```diff
@@ -4,5 +4,6 @@ import amountForServerFormatter from './server';
 export const integerFormatter = formatter( { round: 0 } );
 export const amountInputFormatter = formatter( { round: 2, prefix: '€', integerSeparator: '' } );
 export const millionFormatter = formatter( { round: 1, prefix: '€ ', suffix: 'M', padRight: 1 } );
+export const millionFormatterNumericOnly = formatter( { round: 1, padRight: 1 } );
```
```suggestion
export const millionFormatterNumeric = formatter( { round: 1, padRight: 1 } );
```

moiikana committed:
Archive C23_WMDE_Mobile_EN_01

moiikana committed:
Prepare campaign for C23_WMDE_Mobile_EN_02

- based on EN 01 var
- headline change in both banners
- button label change in both banners
- additional mini banner button in both banners
- both show time of the day

https://phabricator.wikimedia.org/T352638

moiikana committed:
Implement VAR for C23_WMDE_Mobile_EN_02

- new step controller that sets the address type to anonymous in order
to lead the user directly to the payment provider page
- move start date

https://phabricator.wikimedia.org/T352638

Abban commented on banners/mobile_english/content/BannerText.vue:
```diff
@@ -5,24 +5,45 @@
+
+const { currentTime, startTimer, stopTimer } = useCurrentTime( getCurrentTime );
+
+watch( () => props.playLiveText, ( shouldPlay: boolean ) => {
```
This timer shouldn't start until the banner moves to the full page so the watcher should be:

```js
watch( () => props.playLiveText, ( shouldPlay: boolean ) => {
	if ( shouldPlay ) {
		startTimer();
	} else {
		stopTimer();
	}
} );
```

And you can remove `onMounted()`. This will make your tests pass again.

moiikana commented:
this still needs 1-2 additional tests

moiikana committed:
Remove American Express Logo

- the service is currently not supported

gbirke reviewed (commented):
Looks good so far. Ping me when the test are done and I'll approve

gbirke commented on banners/mobile_english/components/MiniBanner.vue:
```diff
@@ -7,21 +7,42 @@
 import CloseIconCircle from '@src/components/Icons/CloseIconCircle.vue';
 
-defineEmits( [ 'showFullPageBanner', 'close' ] );
+defineEmits( [ 'showFullPageBanner', 'showFullPageBannerPreselected', 'close' ] );
```
I find this event name (and the functions for its handler) a bit confusing - *what* is preselected? I'd suggest something like `showFullPageBannerWithAmount` or `showFullPageBannerUsingPreselectedAmount` etc.

gbirke commented on banners/mobile_english/components/BannerVar.vue:
```diff
@@ -155,7 +156,14 @@ function onClose( feature: TrackingFeatureName, userChoice: CloseChoices ): void
+	tracker.trackEvent( new MobileMiniBannerExpandedEvent( 'different-amount' ) );
+}
+
+function onshowFullPageBannerPreselected(): void {
```
This handler is missing a camel case. Please fix when renaming.
